.github/workflows/bot.yml

```diff
@@ -0,0 +1,113 @@
+name: GitHub Bot
+
+on:
+  # Watch for changes on PR state, assignees, labels, head branch and draft/ready status
+  pull_request_target:
+    types:
+      - assigned
+      - unassigned
+      - labeled
+      - unlabeled
+      - opened
+      - reopened
+      - synchronize # PR head updated
+      - converted_to_draft
+      - ready_for_review
+
+  # Watch for changes on PR comment
+  issue_comment:
+    types: [created, edited, deleted]
+
+  # Manual run from GitHub Actions interface
+  workflow_dispatch:
+    inputs:
+      pull-request-list:
+        description: "PR(s) to process: specify 'all' or a comma separated list of PR numbers, e.g. '42,1337,7890'"
+        required: true
+        default: all
+        type: string
+
+jobs:
+  # This job creates a matrix of PR numbers based on the inputs from the various
+  # events that can trigger this workflow so that the process-pr job below can
+  # handle the parallel processing of the pull-requests
+  define-prs-matrix:
+    name: Define PRs matrix
+    # Prevent bot from retriggering itself
+    if: ${{ github.actor != vars.GH_BOT_LOGIN }}
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      pr-numbers: ${{ steps.pr-numbers.outputs.pr-numbers }}
+
+    steps:
+      - name: Parse event inputs
+        id: pr-numbers
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Triggered by a workflow dispatch event
+          if [ '${{ github.event_name }}' = 'workflow_dispatch' ]; then
+
+            # If the input is 'all', create a matrix with every open PRs
+            if [ '${{ inputs.pull-request-list }}' = 'all' ]; then
+              pr_list=`gh pr list --state 'open' --repo '${{ github.repository }}' --json 'number' --template '{{$first := true}}{{range .}}{{if $first}}{{$first = false}}{{else}}, {{end}}{{"\""}}{{.number}}{{"\""}}{{end}}'`
+              [ -z "$pr_list" ] && echo 'Error: no opened PR found' >&2 && exit 1
+              echo "pr-numbers=[$pr_list]" >> "$GITHUB_OUTPUT"
```

Comment:
Should this not be done by the program?

Reply:
yes please, let's have it as a sub-command.

Reply:
Done